test/test_job_retrieval.py

```diff
@@ -183,8 +185,11 @@ def test_jobs_filter_by_program_id(self, service):
 
         job = run_program(service=service, program_id=program_id)
         job_1 = run_program(service=service, program_id=program_id_1)
-        job.wait_for_final_state()
-        job_1.wait_for_final_state()
+        with mock.patch.object(
+            RuntimeJob, "wait_for_final_state", side_effect=time.sleep(0.5)
```

Comment:
Why do we have to put different sleep times in different functions? This can get very flaky. I think we should write a function that returns status in BaseFakeRuntimeJob and any call to wait_for_final_state should be patched with a function that constantly checks status method in BaseFakeRuntimeJob to see if the job status is final.

Reply:
Agree that we should be very carefuly with any types of timeouts in our unit tests. @rathishcholarajan, @kt474 has updated the code. Does that fit your proposal?